src/components/database.rs

```diff
@@ -126,9 +130,65 @@ impl DatabaseComponent {
     pub fn get_connection(db_connection: &Arc<Option<DBConnection>>) -> &DBConnection {
         db_connection.as_ref().as_ref().unwrap()
     }
+
+    pub async fn execute_query(
+        query: Query<'_, Postgres, PgArguments>,
+        transaction: Option<Transaction<'_, Postgres>>,
+        pool: &Pool<Postgres>,
```

Comment:
Is the `pool` not used if a transaction is passed? [Should we use some kind of `Either` instead](https://docs.rs/either/1.1.0/either/enum.Either.html) or some abstraction on top of these two parameters?